小米 6 Global 刷完之后 APP FC

MIUI 版本: MIUI 9 Global 8.4.19
提取固件版本: MIUI 9 China 8.4.19 / 8.5.8
表现: 使用 TWRP 刷入成功之后重启，小米钱包及小米智能卡均无法运行，错误 log 如下:

小米智能卡: https://paste.ubuntu.com/p/gkFwrGDDvx/
小米钱包: https://paste.ubuntu.com/p/yZXMKzhJwW/

小米6欧盟版8.5.8制作刷了钱包和公交卡包以后会报这个错，还缺什么？
<pre>
java.lang.RuntimeException: Unable to instantiate application com.mipay.wallet.App: java.lang.ClassNotFoundException: Didn't find class "com.mipay.wallet.App" on path: DexPathList[[zip file "/system/app/Mipay/Mipay.apk"],nativeLibraryDirectories=[/system/app/Mipay/lib/arm64, /system/fake-libs64, /system/app/Mipay/Mipay.apk!/lib/arm64-v8a, /system/lib64, /system/vendor/lib64, /system/lib64, /system/vendor/lib64]]
	at android.app.LoadedApk.makeApplication(Unknown Source:171)
	at android.app.ActivityThread.handleBindApplication(Unknown Source:1032)
	at android.app.ActivityThread.-wrap1(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(Unknown Source:502)
	at android.os.Handler.dispatchMessage(Unknown Source:21)
	at android.os.Looper.loop(Unknown Source:142)
	at android.app.ActivityThread.main(Unknown Source:71)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Unknown Source:11)
	at com.android.internal.os.ZygoteInit.main(Unknown Source:200)
Caused by: java.lang.ClassNotFoundException: Didn't find class "com.mipay.wallet.App" on path: DexPathList[[zip file "/system/app/Mipay/Mipay.apk"],nativeLibraryDirectories=[/system/app/Mipay/lib/arm64, /system/fake-libs64, /system/app/Mipay/Mipay.apk!/lib/arm64-v8a, /system/lib64, /system/vendor/lib64, /system/lib64, /system/vendor/lib64]]
	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:93)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
	at android.app.Instrumentation.newApplication(Unknown Source:0)
	at android.app.LoadedApk.makeApplication(Unknown Source:68)
	... 9 more
	Suppressed: java.io.IOException: No original dex files found for dex location /system/app/Mipay/Mipay.apk
		at dalvik.system.DexFile.openDexFileNative(Native Method)
		at dalvik.system.DexFile.openDexFile(DexFile.java:353)
		at dalvik.system.DexFile.<init>(DexFile.java:100)
		at dalvik.system.DexFile.<init>(DexFile.java:74)
		at dalvik.system.DexPathList.loadDexFile(DexPathList.java:374)
		at dalvik.system.DexPathList.makeDexElements(DexPathList.java:337)
		at dalvik.system.DexPathList.<init>(DexPathList.java:157)
		at dalvik.system.BaseDexClassLoader.<init>(BaseDexClassLoader.java:65)
		at dalvik.system.PathClassLoader.<init>(PathClassLoader.java:64)
		at com.android.internal.os.PathClassLoaderFactory.createClassLoader(Unknown Source:4)
		at android.app.ApplicationLoaders.getClassLoader(Unknown Source:47)
		at android.app.ApplicationLoaders.getClassLoader(Unknown Source:8)
		at android.app.LoadedApk.createOrUpdateClassLoaderLocked(Unknown Source:230)
		at android.app.LoadedApk.getClassLoader(Unknown Source:6)
		at android.app.LoadedApk.getResources(Unknown Source:26)
		at android.app.ContextImpl.createAppContext(Unknown Source:24)
		at android.app.ActivityThread.handleBindApplication(Unknown Source:701)
		... 8 more
</pre>

--- a/cleaner-fix.sh
+++ b/cleaner-fix.sh
@@ -132,7 +132,13 @@ deodex() {
             fi
         apkfile=$deoappdir/$app/$app.apk
         $sevenzip d "$apkfile" $dexclass >/dev/null
-        $aapt add -fk $apkfile $deoappdir/$app/classes.dex || return 1
+        pushd $deoappdir/$app
+        adderror=false
+        $aapt add -fk $app.apk classes.dex || adderror=true
+        popd
+        if $adderror; then
+            return 1
+        fi
         rm -f $deoappdir/$app/classes.dex
         $zipalign -f 4 $apkfile $apkfile-2 >/dev/null 2>&1
         mv $apkfile-2 $apkfile

--- a/extract.sh
+++ b/extract.sh
@@ -146,7 +146,13 @@ deodex() {
                 fi
             fi
         done
-        $aapt add -fk $deoappdir/$app/$app.apk $deoappdir/$app/$deoarch/classes*.dex || return 1
+        pushd $deoappdir/$app/$deoarch
+        adderror=false
+        $aapt add -fk ../../$app.apk classes*.dex || adderror=true
+        popd
+        if $adderror; then
+            return 1
+        fi
         apkfile=$deoappdir/$app/$app.apk
         $zipalign -f 4 $apkfile $apkfile-2 >/dev/null 2>&1
         mv $apkfile-2 $apkfile
@@ -184,7 +190,13 @@ deodex() {
             return 1
         fi
         $sevenzip d "$apkfile" $dexclass >/dev/null
-        $aapt add -fk $deoappdir/$app/$app.apk $deoappdir/$app/classes*.dex || return 1
+        pushd $deoappdir/$app
+        adderror=false
+        $aapt add -fk $app.apk classes*.dex || adderror=true
+        popd
+        if $adderror; then
+            return 1
+        fi
         rm -f $deoappdir/$app/classes.dex
         $zipalign -f 4 $apkfile $apkfile-2 >/dev/null 2>&1
         mv $apkfile-2 $apkfile

